docs/workspace/setup.md

```diff
@@ -0,0 +1,190 @@
+
+        - [Docker](https://docs.docker.com/get-started/overview/){target=_blank}
+            - [Install Docker Desktop](https://docs.docker.com/engine/install/){target=_blank}
+            - Use the WSL 2 backend for Docker Desktop
+
+    === ":material-apple: MacOS"
+
+        - [Docker](https://docs.docker.com/get-started/overview/){target=_blank}
+            - [Install Docker Desktop](https://docs.docker.com/engine/install/){target=_blank}
+
+    === ":material-linux: Linux"
+
+        - [Docker](https://docs.docker.com/get-started/overview/){target=_blank}
+            - [Install Docker Engine](https://docs.docker.com/engine/install/){target=_blank}
+            - [Manage Docker as a non-root user](https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user){target=_blank}
+            - [Configure Docker to start on boot](https://docs.docker.com/engine/install/linux-postinstall/#configure-docker-to-start-on-boot){target=_blank}
+
+    - [VS Code](https://code.visualstudio.com/){target=_blank}
+        - [Install VS Code](https://code.visualstudio.com/download){target=_blank}
+        - [Install VS Code Remote Development Extension Pack](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack){target=_blank}
+
+2. For Windows and MacOS, additional configuration to run GUI applications
+
+    === ":simple-windows11: Windows 11"
+
+        GUI applications work without additional configuration,
+        *but if you upgraded from Windows 10 make sure to update the WSL kernel:* `wsl --update`
+        in an *administrator* PowerShell window
+
+    === ":material-microsoft-windows: Windows 10"
+
+        Follow [this guide](https://techcommunity.microsoft.com/t5/windows-dev-appconsult/running-wsl-gui-apps-on-windows-10/ba-p/1493242){target=_blank}
+        up to, but not including setting the `DISPLAY` variable. Then, add the following to your `~/.bashrc` of the WSL
+        filesystem:
+
+        ```bash
+        export DISPLAY="`grep nameserver /etc/resolv.conf | sed 's/nameserver //'`:0"
+        ```
+
+        If VS Code is open, then restart it.
+
+    === ":material-apple: MacOS"
+
+        ### XQuartz Configuration
+        Follow [this guide](https://gist.github.com/sorny/969fe55d85c9b0035b0109a31cbcb088){target=_blank} to setup 
+        XQuartz. Then, follow these additional XQuartz configuration steps:
+
+        ```bash
+        # Copy contents to ~/.xinitrc
+        cp /opt/X11/etc/X11/xinit/xinitrc ~/.xinitrc
+
+        # Append "xhost +localhost" to .xinitrc
+        echo "xhost +localhost" >> ~/.xinitrc
+        ```
+
+        If XQuartz is open, then restart it.
+
+        ### Zsh Configuration
+
+        Execute these commands in your terminal:
+
+        ```bash
+        echo 'export MAC_DOCKER_LOCALHOST="docker.for.mac.host.internal"' >> ~/.zshrc
+        echo 'export DISPLAY=:0' >> ~/.zshrc
+        ```
+        If VS Code is open, then restart it.
+
```

Comment:
@patrick-5546 I modified the instructions slightly here. These are probably equivalent, but please check anyways.

Reply:
Looks good